main/src/ca/uwaterloo/flix/api/lsp/provider/CodeActionProvider.scala

```diff
@@ -208,21 +199,26 @@ object CodeActionProvider {
     *   enum Abc { }
     * }}}
     */
-  private def mkNewEnum(name: String, uri: String): CodeAction = CodeAction(
+  private def mkNewEnum(name: String, uri: String, ap: AnchorPosition): CodeAction = {
+    val insertPosition = Position(ap.line, ap.col)
+    val leadingSpaces = " " * ap.spaces
+    CodeAction(
     title = s"Create enum '$name'",
     kind = CodeActionKind.QuickFix,
     edit = Some(WorkspaceEdit(
       Map(uri -> List(TextEdit(
-        Range(Position(1, 1), Position(1, 1)), // TODO: We should figure out where to best place the new enum.
+        Range(insertPosition, insertPosition),
         s"""
-           |enum $name {
+           |${leadingSpaces}enum $name {
```

Comment:
I think a helpful function could be useful. It could be called indent and take a string and an anchor position and then indent the string appropriately (indenting before and for each newline)